dtfs-central-api/src/services/utilisation-report-data-validator/utilisation-report-cell-validators/validate-csv-data/rows/index.ts

```diff
@@ -0,0 +1,56 @@
+import { UtilisationReportCsvRowData, UtilisationReportDataValidationError, UtilisationReportFacilityData } from '@ukef/dtfs2-common';
+import { generateBaseCurrencyErrors } from '../../helpers/generate-base-currency-errors';
+import { generateFacilityUtilisationErrors } from '../../helpers/generate-facility-utilisation-errors';
+
+/**
+ * validateRows
+ * Generate and returns validation errors for each facility in csv rows
+ * if multiple facilities have the same ukef facility ids,
+ * and the "base currency" and "facility utilisation" values do not match
+ * generate validation errors for each facility
+ * Uses a map to store the facility ids and their base currency and facility utilisation values to match against
+ * @param {UtilisationReportCsvRowData[]} csvData - The csv data
+ * @returns {UtilisationReportDataValidationError[]} Generated errors, null if the data is valid
+ */
+export const validateRows = (csvData: UtilisationReportCsvRowData[]): UtilisationReportDataValidationError[] => {
```

Comment:
Was there a comment before that I thought was addressed of making this a bit more of an explicit name?

Reply:
Will discuss